src/dex/hooks/useTabFilters.tsx

```diff
@@ -1,12 +1,11 @@
-import { useState } from "react";
+import { useCallback, useState } from "react";
 
 export function useTabFilters(initialIndex = 0) {
   const [tabIndex, setTabIndex] = useState(initialIndex);
 
-  function handleTabChange(index: number) {
+  const handleTabChange = useCallback((index: number) => {
```

Comment:
Why is `useCallback` needed here?

Reply:
I added the useCallback here, because I was passing this function as a dependency in useEffect for switching tabs baeed on route state. Now I have updated the logic so we dont need it anymore.